### PR TITLE
SWS-910 - De-deprecate the older, interface-based mappings

### PR DESCRIPTION
**Pull request by gregturn**

### CODE REVIEW

File: spring-ws-core/src/main/java/org/springframework/ws/server/endpoint/mapping/PayloadRootQNameEndpointMapping.java [hunk 1]
```diff
@@ -43,9 +43,7 @@
  * @author Arjen Poutsma
  * @see org.springframework.xml.namespace.QNameEditor
  * @since 1.0.0
- * @deprecated as of Spring Web Services 2.0, in favor of {@link PayloadRootAnnotationMethodEndpointMapping}.
```
Comment (artembilan) [inline spring-ws-core/src/main/java/org/springframework/ws/server/endpoint/mapping/PayloadRootQNameEndpointMapping.java]: Thank you @gregturn !
Although I think we should add `@see` from these classes to be flexible.

From other side recall me, please, if you did some changes to Docs on the matter.
Shouldn't we say anything there too?

Comment (gregturn) [inline spring-ws-core/src/main/java/org/springframework/ws/server/endpoint/mapping/PayloadRootQNameEndpointMapping.java]: Regarding @see, good point.

I chatted with @poutsma about docs. Essentially, Spring MVC doesn't document the older way of doing things, and we essentially follow Spring MVC. So we aren't bringing back the doc stuff.

Comment (artembilan) [inline spring-ws-core/src/main/java/org/springframework/ws/server/endpoint/mapping/PayloadRootQNameEndpointMapping.java]: Make sense :+1: 
Feel free to merge from my perspective

**Comment by poutsma** [comment]
Looks good from my end.

**Pull request merged**